js/API/API.js

```diff
@@ -227,16 +227,11 @@ data - The data to download.
 */
 
 function downloadFile(fileName, data) {
-    var a = document.createElement("a");
-    document.body.appendChild(a);
-	
-	var blob = new Blob([data], {type: "octet/stream"}),
-	    url = window.URL.createObjectURL(blob);
-	a.href = url;
-	a.download = fileName;
-	a.click();
-	window.URL.revokeObjectURL(url);
-	a.remove();
+	var downloadlink = document.body.appendChild(document.createElement("a"));
+	downloadlink.download = fileName;
+	downloadlink.href = "data:text/plain;base64," + btoa(unescape(encodeURIComponent(data)));
```

Comment:
Hi! I now figured out what was wrong with the original code in Firefox. Firefox removed the ObjectURL before the link was pressed, which made the download not work.

To correct this I took your original function and rewrote the clean-up part to 
    setTimeout(function() {
		window.URL.revokeObjectURL(url);
		a.remove();
	},1);

This makes the Javascript engine process all the pending events before it cleans up the ObjectURL

Have a download, and try it out in FF and Chrome so you can see it works! But I think the code is ready for merging now